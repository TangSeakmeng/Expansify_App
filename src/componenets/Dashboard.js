import React from 'react';

import ExpenseList from './ExpensesList';
import ExpenseListFilter from './ExpenseListFilter';


const ExpenseDashboardPage = () => (
    <div>
        <ExpenseListFilter />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;