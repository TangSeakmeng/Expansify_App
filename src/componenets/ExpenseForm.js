import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

// const date = new Date();
const now = moment().format('MMM Do, YYYY');

export default class ExpenseForm extends React.Component {
    constructor(props) {
        super(props);
        
        this.state = {
            description: props.expense ? props.expense.description : '',
            note: props.expense ? props.expense.note : '',
            amount: props.expense ? props.expense.amount : 0,
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            calendarFocused: false,
            error: ''
        };
    }
    

    // state = {
    //     description: '',
    //     note: '',
    //     amount: 0,
    //     createdAt: moment(),
    //     calendarFocused: false,
    //     error: ''
    // };

    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description } ));
    };

    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState(() => ({ note }));
    };

    onAmountChange = (e) => {
        const amount = e.target.value;
        if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({ amount }));
        }
    }

    onDateChange = (createdAt) => {
        if(createdAt) {
            this.setState(() => ({ createdAt }));
        }
    }

    onFocusChange = ({ forcused }) => {
        this.setState((prevState) => ({ calendarFocused: !prevState.calendarFocused }));
    }

    onSubmit = (e) => {
        e.preventDefault();
        
        if(!this.state.description || !this.state.amount) {
            this.setState(() => ({ error: 'Invalid Input!' }))

        } else {
            this.setState(() => ({ error: '' }))
            this.props.onSubmit({
                description: this.state.description,
                // amount: parseFloat(this.state.amount, 10) * 100, 
                amount: parseFloat(this.state.amount, 10), 
                createdAt: this.state.createdAt.valueOf(),
                note: this.state.note
            })
        }
    }

    render() {
        return (
            <form onSubmit={this.onSubmit} className="form">
                {this.state.error && <p className="form__error">{this.state.error}</p>}
                <input type="text" placeholder="description" autoFocus value={this.state.description} onChange={this.onDescriptionChange} className="text-input" />
                <input type="number" placeholder="amount" autoFocus value={this.state.amount} onChange={this.onAmountChange} className="text-input" />
                <SingleDatePicker 
                    date={this.state.createdAt} 
                    onDateChange={this.onDateChange} 
                    focused={this.state.calendarFocused} 
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1} 
                    isOutsideRange={() => false}
                />
                <textarea placeholder="add a note for your expense (optional)" value={this.state.note} onChange={this.onNoteChange} className="textarea"></textarea>
                <div>
                    <button className="button">{ this.props.buttonTitle }</button>
                </div>
            </form>
        );
    };
};