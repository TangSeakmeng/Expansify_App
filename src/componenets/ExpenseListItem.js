import React from 'react';
import moment from 'moment';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';

import { removeExpense } from '../actions/expenses';

const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{ description }</h3>
        </Link>
        <p>{ numeral(amount).format('$0,0.00') } - { moment(createdAt).format('MMMM Do, YYYY') }</p>
        
        {/* <button onClick={() => {
            props.dispatch(removeExpense({ id }))
        }}>Remove</button> */}
    </div>
);

export default ExpenseListItem;
