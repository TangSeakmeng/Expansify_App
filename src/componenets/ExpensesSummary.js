import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import expenses_total from '../selectors/expenses-total';
import selectorExpenses from '../selectors/expenses';

const ExpenseSummary = (props) => {
    const expenseWord = props.expenses.length <= 1 ? 'expense' : 'expenses';

    return (    
        <div>
            <p>Viewing {props.expenses.length} {expenseWord} totalling { numeral(expenses_total(props.expenses)).format('$0,0.00') }</p>
        </div>
    )
};

const mapStateToProps = (state) => {
    return {
        expenses: selectorExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseSummary);