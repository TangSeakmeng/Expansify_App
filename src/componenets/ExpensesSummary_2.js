// import React from 'react';
// import { connect } from 'react-redux';
// import numeral from 'numeral';
// import expenses_total from '../selectors/expenses-total';
// import selectorExpenses from '../selectors/expenses';

// const ExpenseSummary = ({ expenseCount, expenseTotal }) => {
//     const expenseWord = expenseCount <= 1 ? 'expense' : 'expenses';
//     const total = numeral(expenseTotal).format('$0,0.00')

//     return (    
//         <div>
//             <h1>Viewing {expenseCount} {expenseWord} totalling {total}</h1>
//         </div>
//     )
// };

// const mapStateToProps = (state) => {
//     const visibleExpenses = selectorExpenses(state.expenses, state.filters);

//     return {
//         expenseCount: visibleExpenses.length,
//         expenseTotal: expenses_total(visibleExpenses)
//     };
// };

// export default connect(mapStateToProps)(ExpenseSummary);

// ----------------------------------------------------------------------------------------

import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses' ;
  const formattedExpensesTotal = numeral(expensesTotal).format('$0,0.00');
  
  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span></h1>
        <div className="page-header__action">
          <Link to="/create" className="button">Add Expense</Link>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);

  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
