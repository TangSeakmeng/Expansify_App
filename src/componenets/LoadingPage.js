import React from 'react';

const LoadingPage = () => (
    <div className="loading">
        <img src="/images/loader.gif" className="loading__image"></img>
    </div>
);

export default LoadingPage;