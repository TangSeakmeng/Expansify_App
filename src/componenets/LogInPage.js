import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';
import { firebase, googleAuthProvider } from '../firebase/firebase';

// export const LogInPage = ({ startLogin }) => (
//     <div>
//         <button onClick={() => firebase.auth().signInWithPopup(googleAuthProvider)}>Login</button>
//     </div>
// );

// export const LogInPage = ({ startLogin }) => (
//     <div>
//         <button onClick={() => startLogin }>Login</button>
//     </div>
// );

// export class LogInPage extends React.Component {
//     onClick = () => {
//         this.props.startLogin();
//         this.props.history.push('/dashboard');
//     }

//     render() {
//         return (
//             <button onClick={this.onClick}>Login</button>
//         );
//     };
// };

export const LogInPage = ({ startLogin }) => (
    <div>
        <button onClick={ startLogin }>Login</button>
    </div>
);
  
const mapDispatchToProps = (dispatch) => {
    return {
        startLogin: () => {
            dispatch(startLogin())
        }
    }
};
  
export default connect(undefined, mapDispatchToProps)(LogInPage);