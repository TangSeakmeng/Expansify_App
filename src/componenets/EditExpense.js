import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpenses } from '../actions/expenses';

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/dashboard');
    }

    onClick = () => {
        this.props.startRemoveExpenses({ id: this.props.expense.id })
        this.props.history.push('/dashboard');
    }

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm expense={this.props.expense} buttonTitle='Edit Expense' onSubmit={this.onSubmit} />
                    <button onClick={this.onClick} className="button--secondary">Remove Expense</button>
                </div>
            </div>
        );
    };
};

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => {
            return expense.id === props.match.params.id;
        })
    };
};

const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => {
        dispatch(startEditExpense(id, expense))
    },
    startRemoveExpenses: (data) => {
        dispatch(startRemoveExpenses(data))
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);