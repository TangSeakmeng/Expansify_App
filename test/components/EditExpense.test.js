import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../src/componenets/EditExpense';
import expenses from '../fixtures/arr';

let startEditExpense, startRemoveExpenses, history, wrapper;

beforeEach(() => {
    startEditExpense = jest.fn();
    startRemoveExpenses = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<EditExpensePage startEditExpense={startEditExpense} startRemoveExpenses={startRemoveExpenses} history={history} expense={expenses[0]} />);
});

test('should render EditExpensePage Correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle on editExpense', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[0]);
    expect(history.push).toHaveBeenLastCalledWith('/dashboard');
    expect(startEditExpense).toHaveBeenLastCalledWith(expenses[0].id, expenses[0]);
})

test('should handle on removeExpense', () => {
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenLastCalledWith('/dashboard');
    expect(startRemoveExpenses).toHaveBeenLastCalledWith({ id: expenses[0].id });
})
