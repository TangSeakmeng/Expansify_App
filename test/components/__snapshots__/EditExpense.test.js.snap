// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage Correctly 1`] = `
<div>
  <h1>
    Edit Expense
  </h1>
  <ExpenseForm
    buttonTitle="Edit Expense"
    expense={
      Object {
        "amount": 195,
        "createdAt": 0,
        "description": "Water Bill",
        "id": 1,
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
